pkg/features/kube_features.go

```diff
@@ -522,7 +524,7 @@ var defaultKubernetesFeatureGates = map[featuregate.Feature]featuregate.FeatureS
 	ExpandPersistentVolumes:        {Default: true, PreRelease: featuregate.Beta},
 	ExpandInUsePersistentVolumes:   {Default: true, PreRelease: featuregate.Beta},
 	ExpandCSIVolumes:               {Default: true, PreRelease: featuregate.Beta},
-	AttachVolumeLimit:              {Default: true, PreRelease: featuregate.Beta},
+	AttachVolumeLimit:              {Default: true, PreRelease: featuregate.GA, LockToDefault: true}, // remove in 1.19
```

Comment:
I'm a bit confused. I noticed that we set `LockToDefault` to `true` in `CSINodeInfo` as well, but if I understand the docs [1] correctly, these features are CLI arguments and they should work for at least one version:

```
Rule #5b: CLI elements of admin-facing components (e.g. kubelet) must function after their announced deprecation for no less than:

    GA: 6 months or 1 release (whichever is longer)
```

Shouldn't these features be deprecated (keep working with a warning) in this version and only locked in 1.18?

CC @msau42 @liggitt 

[1] https://kubernetes.io/docs/reference/using-api/deprecation-policy/#deprecating-a-flag-or-cli

Reply:
see feature-gate-specific lifecycle in [section 7](https://kubernetes.io/docs/reference/using-api/deprecation-policy/#deprecating-a-feature-or-behavior):

> An exception to the above rule is feature gates. Feature gates are key=value pairs that allow for users to enable/disable experimental features.
> 
> Feature gates are intended to cover the development life cycle of a feature - they are not intended to be long-term APIs. As such, they are expected to be deprecated and removed after a feature becomes GA or is dropped.
> 
> As a feature moves through the stages, the associated feature gate evolves. The feature life cycle matched to its corresponding feature gate is:
> ...
> GA: the feature gate is deprecated (see “Deprecation”) and becomes non-operational.